presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFileFooterReader.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.pagefile;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static java.lang.Math.toIntExact;
+import static java.nio.ByteBuffer.allocate;
+import static java.nio.ByteOrder.LITTLE_ENDIAN;
+import static java.util.Objects.requireNonNull;
+
+public class PageFileFooterReader
+{
+    private long[] stripeOffsets;
+    private ByteBuffer readBuffer = allocate(SIZE_OF_LONG);
```

Comment:
You can wrap the input stream with the `InputStreamSliceInput` and use the `#getLong` method that reads longs in LITTLE_ENDIAN